contributors/design-proposals/api-machinery/admission-control-webhooks.md

```diff
@@ -211,30 +203,41 @@ When a webhook is persistently failing to allow e.g. pods to be created, then th
+It is possible that two different webhooks want to access the same
+stored-resource via different GVKs.  In this case there could be extra
+conversions.  Conversion does take some CPU.  However, we don't expect this to be
+a significant issue for beta because:
 
-It's TBD if we go with Plan A or B. 
+*   Pods are by far the most common core object to webhook, based on the  [Use Cases](#use-cases-detailed-descriptions), and pods do not have multiple versions.  Most or all other resources we know we plan to Webhook also do not have multiple versions.  So extra conversions will likely not, in practice, be performed for the majority of webhook calls.
+*   Reads are ~10x more common that writes in a typical cluster, so any increase in compute cost for writes to multi-version objects is significantly mitigated.
+*   Most objects have < 3 versions; webhooks can be ordered any way at the moment, so they could be called in the manner which minimizes the number of conversions. IOW, conversions should worst-case scale with the number of versions, not the number of webhooks.
 
-It is possible that two different webhooks want to access the same stored-resource via different GVKs.  In this case there could be extra conversions.  Conversion do take some CPU.  However, we don't expect this to be a significant issue for beta because:
+If the administrator chooses to install webhooks that all use the same GVKs for a given internal type, then it will be more efficient, since the apiserver may not need to convert every time: it just needs to patch the json, and then send it back out again without converting to internal.  
 
+## Registering for all possible representations of the same object
 
+Some Kubernetes resources are mounted in the api type system at multiple places
+(e.g., during a move between groups). Additionally, some resources have multiple
+active versions. There's not currently a way to easily tell which of the exposed
+resources map to the same "storage location". We will not try to solve that
+problem at the moment: if the system administrator wishes to hook all
+deployments, they must (e.g.) make sure their hook is registered for both
+deployments.v1beta1.extensions AND deployments.v1.apps.
 
-*   Pods are by far the most common core object to webhook, based on the  [Use Cases](#use-cases-detailed-descriptions), and pods do not have multiple versions.  Most or all other resources we know we plan to Webhook also do not have multiple versions.  So extra conversion is not needed in practice.
-*   Reads are ~10x more common that writes in a typical cluster, so any increase in compute cost for writes to multi-version objects is significantly mitigated.
+This is likely to be error-prone, especially over upgrades. For GA, we may
+consider mechanisms to make this easier. One idea which has been floated in
```

Comment:
I'd clip the "One idea which has been floated...." sentence.  Off the cuff, I'd do it differently and I'd rather not get bogged down on it here.

Reply:
sigh, ok